src/cleanup.ts

```diff
@@ -29,4 +47,82 @@ function addCachedCliToPath(): boolean {
     return true;
 }
 
+interface BuildPublishResponse {
+    modules: any[];
+}
+
+async function hasUnpublishedModules(workingDirectory: string): Promise<boolean> {
+    // Save the old value of the environment variable to revert it later
+    const origValue: string | undefined = process.env[Utils.JFROG_CLI_COMMAND_SUMMARY_OUTPUT_DIR_ENV];
+    try {
+        // Avoid saving a command summary for this dry-run command
+        core.exportVariable(Utils.JFROG_CLI_COMMAND_SUMMARY_OUTPUT_DIR_ENV, '');
+
+        // Running build-publish command with a dry-run flag to check if there are any unpublished modules, 'silent' to avoid polluting the logs
+        const responseStr: string = await Utils.runCliAndGetOutput(['rt', 'build-publish', '--dry-run'], { silent: true, cwd: workingDirectory });
+
+        // Parse the JSON string to an object
+        const response: BuildPublishResponse = JSON.parse(responseStr);
+        // Check if the "modules" key exists and if it's an array with more than one item
+        return response.modules != undefined && Array.isArray(response.modules) && response.modules.length > 0;
+    } catch (error) {
+        core.error('Failed to parse JSON: ' + error);
+        return false; // Return false if parsing fails
+    } finally {
+        core.exportVariable(Utils.JFROG_CLI_COMMAND_SUMMARY_OUTPUT_DIR_ENV, origValue);
+    }
+}
+
+async function collectAndPublishBuildInfoIfNeeded() {
+    const workingDirectory: string = getWorkingDirectory();
+    // Check if there are any unpublished modules
+    if (!(await hasUnpublishedModules(workingDirectory))) {
+        return;
+    }
+
+    core.startGroup('Collect environment variables information');
+    await Utils.runCli(['rt', 'build-collect-env'], { cwd: workingDirectory });
+    core.endGroup();
+
+    core.startGroup('Collect the Git information');
+    await Utils.runCli(['rt', 'build-add-git'], { cwd: workingDirectory });
```

Comment:
This will fail if there is no Checkout step or if the git repository has been deleted:
> 15:03:46 [🚨Error] Could not find .git